ott/core/potentials.py

```diff
@@ -0,0 +1,164 @@
+import abc
+from typing import Any, Callable, Dict, Sequence, Tuple, Union
+
+import jax
+import jax.numpy as jnp
+import jax.tree_util as jtu
+
+from ott.geometry import pointcloud
+
+__all__ = ["EntropicMap", "DualPotentials"]
+Potential_t = Union[jnp.ndarray, Callable[[jnp.ndarray], float]]
+
+
+@jtu.register_pytree_node_class
+class BaseDualPotentials(abc.ABC):
+  """Base class holding the Kantorovich dual potentials.
+
+  Args:
+    f: The first dual potential.
+    g: The second dual potential.
+
+  Notes:
+    Both potentials can be represented either as an :class:`jax.numpy.ndarray`
```

Comment:
Potentials can be provided as a :class:`jax.numpy.ndarray` (in which they are assumed to be tied to sample points identified in a :class:`geometry.Geometry` class) or more generally a function taking :class:`jax.numpy.ndarray` and returning :class:`float`.

Reply:
Removed this, `DualPotentials` now always expect a function. For `EntropicMap`, it has a class method called `from_potentials`.